### ISSUE TITLE
[Camera] Don't force 16:9 aspect as preview size

### ISSUE THREAD

**Issue by ValYouW** [flutter/flutter#80393]
Hi,

**On Android:**
I am trying to achieve a camera preview that is best for taking pictures, not videos (like the native camera app).
Currently it seems that preview size is only limited to 16:9 aspect ratio, as in `CameraUtils::getBestAvailableCamcorderProfileForResolutionPreset` the `ResolutionPreset` is translated only to `CamcorderProfile` values, which are all 16:9.
Having 16:9 previews automatically "crops" the image and the user can't enjoy the fully "zoomed-out" view (see attached images below).

Having an API like in CameraX could be very nice:
```kotlin
Preview.Builder()
  .setTargetAspectRatio(AspectRatio.RATIO_4_3)
  .build()
```

Is there an option to achieve that currently? a workaround? 
I tested by changing the method `CameraUtils::computeBestPreviewSize` to return `new Size(1024, 768)` and indeed I get the fully "zoomed-out" preview...

Here is the current preview I get:
![prev_16_9](https://user-images.githubusercontent.com/4539636/114624816-f0564300-9cb9-11eb-8dc2-0368f4b85d8e.png)

And here is the preview when I change preview size to `Size(1024, 768)`
![prev_4_3](https://user-images.githubusercontent.com/4539636/114624826-f51af700-9cb9-11eb-9230-bdb5fd76e7ec.png)

Thx!

**Comment by darshankawar** [comment]
@ValYouW 
Thanks for the details. I see this as a proposal to not force the aspect ratio for camera preview to be 16:9 always. Although I see couple of related / similar issues which I would like you to go through and confirm if they resemble your case:

https://github.com/flutter/flutter/issues/63358
https://github.com/flutter/flutter/issues/15953

**Comment by ValYouW** [comment]
Hi, I looked at these two issues before, of I understood correctly they only refer to the view widget aspect ratio. Maybe the title of my issue is not accurate, essentially my issue is that I can't get the full image area, it is always cropped.
As can be seen in the images I uploaded, the view widget is 4:3, that's not a problem, but the image is cropped...

**Comment by migalv-fu** [comment]
Yeah, this is something that is hapening to me. I would like to set the 3:4 aspect ration but I can't find anywhere to set this.

**Comment by migalv** [comment]
Yeah, this is something that is hapening to me. I would like to set the 3:4 aspect ration but I can't find anywhere to set this.

**Comment by FourwingsY** [comment]
when ResolutionPreset is `low` or `max`, It returns 4:3 images.
with other settings, it returns 16:9 images.

**Comment by TheManuz** [comment]
> when ResolutionPreset is `low`, `medium (in iOS)` or `max`, It returns 4:3 images. with other settings, it returns 16:9 images.

Ok, this can be a workaround, but obviously not a solution.

1. This aspect ratio behavior is not documented.
2. Two different parameters are mixed together: resolution and aspect ratio. As a result, some resolutions cannot use 4:3, others cannot use 16:9.
